cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -1,23 +1,45 @@
 set(ARROW_ROOT ${CMAKE_BINARY_DIR}/arrow)
 
-set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
-                     " -DARROW_WITH_ZSTD=OFF"
-                     " -DARROW_WITH_BROTLI=OFF"
-                     " -DARROW_WITH_SNAPPY=OFF"
-                     " -DARROW_WITH_ZLIB=OFF"
-                     " -DARROW_BUILD_STATIC=ON"
-                     " -DARROW_BUILD_SHARED=OFF"
-                     " -DARROW_BOOST_USE_SHARED=ON"
-                     " -DARROW_BUILD_TESTS=OFF"
-                     " -DARROW_TEST_MEMCHECK=OFF"
-                     " -DARROW_BUILD_BENCHMARKS=OFF"
-                     " -DARROW_IPC=ON"
-                     " -DARROW_COMPUTE=OFF"
-                     " -DARROW_GPU=OFF"
-                     " -DARROW_JEMALLOC=OFF"
-                     " -DARROW_BOOST_VENDORED=OFF"
-                     " -DARROW_PYTHON=OFF"
-                     " -DARROW_TENSORFLOW=ON") # enable old ABI for C/C++
+if(NOT CMAKE_CXX11_ABI)
+    message(STATUS "ARROW: Disabling the GLIBCXX11 ABI")
+    set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
+                         " -DARROW_WITH_ZSTD=OFF"
+                         " -DARROW_WITH_BROTLI=OFF"
+                         " -DARROW_WITH_SNAPPY=OFF"
+                         " -DARROW_WITH_ZLIB=OFF"
+                         " -DARROW_BUILD_STATIC=ON"
+                         " -DARROW_BUILD_SHARED=OFF"
+                         " -DARROW_BOOST_USE_SHARED=ON"
+                         " -DARROW_BUILD_TESTS=OFF"
+                         " -DARROW_TEST_MEMCHECK=OFF"
+                         " -DARROW_BUILD_BENCHMARKS=OFF"
+                         " -DARROW_IPC=ON"
+                         " -DARROW_COMPUTE=OFF"
+                         " -DARROW_GPU=OFF"
+                         " -DARROW_JEMALLOC=OFF"
+                         " -DARROW_BOOST_VENDORED=OFF"
+                         " -DARROW_PYTHON=OFF"
+                         " -DARROW_TENSORFLOW=ON") # enable old ABI for C/C++
```

Comment:
May want to see if we can swap only the string in the future for code clarity, but this is fine for now.

Reply:
`ARROW_CMAKE_ARGS` is a list of strings. I could remove the `" -DARROW_TENSORFLOW={ON/OFF}"` and use `list(APPEND ARROW_CMAKE_ARGS " -DARROW_TENSORFLOW={ON/OFF}"` as a function of the value `CMAKE_CXX11_ABI`.

Reply:
@mt-jones please follow up on this.

Reply:
This has been done. Merged into PR https://github.com/rapidsai/cudf/pull/668